Fix default filename preprocessing for Windows

fusion1110 opened the pull request:
Problem:
When creating a new file in Xournal++ on Windows (without a PDF background), the suggested filename is empty if the default filename contains %F, %T, or %V (e.g., %F-Note-%H-%M). This prevents users from saving the file unless they manually enter a name.

The root cause is that MinGW-w64 does not fully support strftime formatting codes like %F, %T, or %V (ISO 8601), which results in an empty string when generating the suggested filename.

Solution:
Replaced unsupported strftime codes with equivalents supported on Windows:

%F → %Y-%m-%d
%T → %H-%M-%S
%V → %U

Added a preprocessing function that converts the default filename format to a Windows-compatible version before generating the suggested filename.
Ensures forbidden characters are replaced and the filename is never empty.

Testing:
Tested on Windows 10 using a freshly compiled Xournal++ build from master.
Verified that new files now have a suggested filename according to the default pattern, e.g., 2025-08-25-Note-15-30.

Additional context:
Beginner contribution: This is my first pull request, and it’s not my last!
I really appreciate the maintainers and all contributors for this amazing app that I use daily.

rolandlo commented:
Thanks for the contribution. The issue you pointed out (#4326) is certainly something that needs to be fixed. It already exists in the release branch. Could you please rebase on the `release-1.2` branch (as follows)?

```term
git rebase --onto=upstream/release-1.2 HEAD~ HEAD
```
Then after force-pushing you can change the target branch of the pull request via the Edit button on the top right of this page in the Github interface.

rolandlo commented on src/core/model/Document.cpp:
```diff
@@ -136,6 +136,26 @@ auto Document::createSaveFoldername(const fs::path& lastSavePath) const -> fs::p
 }
 
+
+static std::string preprocessFormatString(const std::string& formatStr) {
+    std::string processed = formatStr;
```
Since the string is copied anyway it would make more sense to pass it by value (and continue to work with `formatStr`)

fusion1110 commented:
> Thanks for the contribution. The issue you pointed out (#4326) is certainly something that needs to be fixed. It already exists in the release branch. Could you please rebase on the `release-1.2` branch (as follows)?
> 
> ```
> git rebase --onto=upstream/release-1.2 HEAD~ HEAD
> ```
> 
> Then after force-pushing you can change the target branch of the pull request via the Edit button on the top right of this page in the Github interface.

Rebased onto release-1.2, force-pushed, and updated the PR target branch, made the suggested changes.
 Ready for review!
